src/tests/ascent/t_ascent_rover.cpp

```diff
@@ -319,6 +349,237 @@ TEST(ascent_rover, test_xray_blueprint_braid_rotated)
+
+    if (is_vtkm_disabled())
+    {
+        return; // Returning early is equivalent to passing the test
+    }
+
+    // Test names
+    const std::string query_name = "tout_rover_xray_blueprint_braid_uniform_multi_domain";
+    const std::string query_ext_name = "_000000.cycle_000000.root";
+
+    // Setup paths
+    const std::string output_path = prepare_output_dir();
+    const std::string query_path = conduit::utils::join_file_path(output_path, 
+                                                                  query_name);
+    const std::string output_data_path = query_path + query_ext_name;
+
+    // Remove old test image
+    const int cycle = 0;
+    remove_test_image(query_path, cycle);
+
+    // Generate and verify test data
+    Node test_data;
+    get_valid_multi_domain_test_data(test_data, 2);
+
+    // Define Ascent actions
+    Node extracts;
+    extracts["e1/type"] = "xray";
+    extracts["e1/params/rover/absorption"] = "radial";
+    extracts["e1/params/rover/emission"] = "radial";
+    extracts["e1/params/rover/filename"] = query_path;
+    extracts["e1/params/rover/output_type"] = "json";
+    extracts["e1/params/rover/precision"] = "double";
+    extracts["e1/params/rover/unit_scalar"] = 1.234f;
+
+    Node actions;
+    Node &add_extracts = actions.append();
+    add_extracts["action"] = "add_extracts";
+    add_extracts["extracts"] = extracts;
+
+    // Execute Ascent actions
+    execute_ascent(test_data, actions);
+
+    // Load and verify output mesh
+    Node xray_blueprint_output, verify_info;
+    load_and_verify_local_data(xray_blueprint_output, output_data_path);
+    Node &state_output = xray_blueprint_output["domain_000000/state"];
+
+    // Load and verify baseline data
+    const std::string yaml = R"yaml(
+        time: 3.1414999961853
+        cycle: 0
+        xray_view: 
+            position: [10.0, 0.0, 48.9897956848145]
+            zoom: 1.0
+            look_at: [10.0, 0.0, 0.0]
+            up: [0.0, 1.0, 0.0]
+            fov: 60.0
+            xpan: 0.0
+            ypan: 0.0
+            near_plane: 4.89897966384888
+            far_plane: 489.89794921875
+        xray_query: 
+            background_intensity: 0.0
+            divide_emis_by_absorb: "false"
+            enable_rays_mesh: "false"
+            height: 200
+            precision: "double"
+            width: 200
+            unit_scalar: 1.23399996757507
+            absorption: "radial"
+            emission: "radial"
+            filename: ""
+            output_type: "json"
+        xray_data: 
+            detector_width: 5.65685440236809
+            detector_height: 5.65685440236809
+            intensity_max: 213.735064037837
+            intensity_min: 0.0
+            optical_depth_max: 3683.56120526528
+            optical_depth_min: 0.0
+            image_topo_order_of_domain_variables: "xyz"
+        domain_id: 0
+    )yaml";
+
+    Node baseline_data;
```

Comment:
I wish we could do the yaml differencing in a helper. A bunch of these have the exact same yaml so it shouldn't be hard to refactor out, but we could go a step further and generate the correct yaml for each test like we do in the visit tests:

https://github.com/visit-dav/visit/blob/84169e780964ebabbddc976a23ba40054c7d5dc9/src/test/tests/queries/xrayimage.py#L564-L573

We are passing in the correct expected values to the strings that then are parsed as conduit nodes for the diff. It should be possible to do the same pattern in C++.

Reply:
yes, I like that idea